File: 56-MergeIntervals/56-MergeIntervals.cpp
// Last updated: 16/10/2025, 04:30:35
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        vector<vector<int>>mergeIntervals;
        if(intervals.size() == 0){
            return mergeIntervals;
        }
        sort(intervals.begin(),intervals.end());
        vector<int>tempInterval = intervals[0];
        for(auto x : intervals){
            if(x[0] <= tempInterval[1]){
                tempInterval[1] = max(x[1],tempInterval[1]);
            }
            else{
                mergeIntervals.push_back(tempInterval);
                tempInterval=x;
            }
        }
        mergeIntervals.push_back(tempInterval);
        return mergeIntervals;
        
    }
};

File: 56-MergeIntervals/56-MergeIntervals.java
// Last updated: 16/10/2025, 04:30:28
class Solution {
    public int[][] merge(int[][] intervals) {
        Arrays.sort(intervals, (a,b) -> Integer.compare(a[0],b[0]));

        int [] temp = intervals[0];
        List<int[]> mergedIntervals = new ArrayList<>();
        for(int[] x : intervals){
            if(x[0] <= temp[1]){
                temp[1] = Math.max(temp[1], x[1]);
            }else{
                mergedIntervals.add(temp);
                temp = x;
            }
        }
        mergedIntervals.add(temp);
        return mergedIntervals.toArray(new int[mergedIntervals.size()][2]);
    }
}